crates/bevy_pbr/src/prepass/mod.rs

```diff
@@ -92,27 +94,65 @@ where
             Shader::from_wgsl
         );
 
-        let Ok(render_app) = app.get_sub_app_mut(RenderApp) else {
-            return;
-        };
+        app.add_system(update_previous_view_projections.in_base_set(CoreSet::PreUpdate))
+            .add_system(update_mesh_previous_global_transforms.in_base_set(CoreSet::PreUpdate));
+
+        let Ok(render_app) = app.get_sub_app_mut(RenderApp) else { return };
 
         render_app
             .add_system_to_schedule(ExtractSchedule, extract_camera_prepass_phase)
             .add_system(prepare_prepass_textures.in_set(RenderSet::Prepare))
+            .add_system(prepare_previous_view_projection_uniforms.in_set(RenderSet::Prepare))
             .add_system(queue_prepass_view_bind_group::<M>.in_set(RenderSet::Queue))
             .add_system(queue_prepass_material_meshes::<M>.in_set(RenderSet::Queue))
             .add_system(sort_phase_system::<Opaque3dPrepass>.in_set(RenderSet::PhaseSort))
             .add_system(sort_phase_system::<AlphaMask3dPrepass>.in_set(RenderSet::PhaseSort))
             .init_resource::<PrepassPipeline<M>>()
             .init_resource::<DrawFunctions<Opaque3dPrepass>>()
             .init_resource::<DrawFunctions<AlphaMask3dPrepass>>()
+            .init_resource::<PreviousViewProjectionUniforms>()
             .init_resource::<PrepassViewBindGroup>()
             .init_resource::<SpecializedMeshPipelines<PrepassPipeline<M>>>()
             .add_render_command::<Opaque3dPrepass, DrawPrepass<M>>()
             .add_render_command::<AlphaMask3dPrepass, DrawPrepass<M>>();
     }
 }
 
+#[derive(Component, ShaderType, Clone)]
+pub struct PreviousViewProjection {
+    pub view_proj: Mat4,
+}
+
+pub fn update_previous_view_projections(
+    mut commands: Commands,
+    query: Query<(Entity, &ExtractedView), (With<Camera3d>, With<VelocityPrepass>)>,
```

Comment:
the previous view proj isn't updating properly, there's never an `ExtractedView` in the main world.

Reply:
Good catch, should be fixed now.